static/skin/index.js

```diff
@@ -1,71 +1,75 @@
-    const mediaCount = getInnerHtml(book, 'mediaCount');
+    function generateBookHtml(book) {
+        const link = book.querySelector('link').getAttribute('href');
+        const title =  getInnerHtml(book, 'title');
+        const description = getInnerHtml(book, 'summary');
+        const id = getInnerHtml(book, 'id');
+        const iconUrl = getInnerHtml(book, 'icon');
+        const articleCount = getInnerHtml(book, 'articleCount');
+        const mediaCount = getInnerHtml(book, 'mediaCount');
 
-    return `<a href='${link}' data-id='${id}'><div class='book'>
-        <div class='book__background' style="background-image: url('${iconUrl}');">
-        <div class='book__title' title='${title}'>${title}</div>
-        <div class='book__description' title='${description}'>${description}</div>
-        <div class='book__info'>${articleCount} articles, ${mediaCount} medias</div>
-        </div>
-    </div></a>`;
-}
+        return `<a href='${link}' data-id='${id}'><div class='book'>
+            <div class='book__background' style="background-image: url('${iconUrl}');">
+            <div class='book__title' title='${title}'>${title}</div>
+            <div class='book__description' title='${description}'>${description}</div>
+            <div class='book__info'>${articleCount} articles, ${mediaCount} medias</div>
+            </div>
+        </div></a>`;
+    }
 
-async function loadAndDisplayBooks() {
-    isFetching = true;
-    fetch(queryUrlBuilder()).then(async (resp) => {
-        const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
-        const books = data.querySelectorAll('entry');
-        let bookHtml = '';
-        books.forEach((book) => {bookHtml += generateBookHtml(book)});
-        document.querySelector('.book__list').innerHTML += bookHtml;
-        incrementalLoadingParams['start'] = incrementalLoadingParams['start'] + incrementalLoadingParams['count'];
-        if (!books.length) {
-            incrementalLoadingParams['count'] = 0;
+    async function loadAndDisplayBooks() {
+        isFetching = true;
+        fetch(queryUrlBuilder()).then(async (resp) => {
+            const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
+            const books = data.querySelectorAll('entry');
+            let bookHtml = '';
+            books.forEach((book) => {bookHtml += generateBookHtml(book)});
+            document.querySelector('.book__list').innerHTML += bookHtml;
+            incrementalLoadingParams.start += incrementalLoadingParams.count;
+            if (books.length < incrementalLoadingParams.count) {
+                incrementalLoadingParams.count = 0;
+            }
+            isFetching = false;
+        });
+    }
+
+    window.addEventListener('resize', async () => {
+        if (isFetching || !incrementalLoadingParams.count || incrementalLoadingParams.count > viewPortToCount()) return;
```

Comment:
What is the justification for the `incrementalLoadingParams.count > viewPortToCount()` check?

Reply:
to check if the screen size is reduced no new data should be fetched in that case until and unless we are at the bottom of the page.

Reply:
My point is that the `incrementalLoadingParams.count > viewPortToCount()` check is redundant. The bottom-of-the-page test is enough, isn't it? If so, then handling of resize and scroll events becomes identical, and then introducing another function is justified.

Reply:
> The bottom-of-the-page test is enough, isn't it?

Though this depends on when the resize event fires - before or after the layout is updated.

Reply:
oh sorry, I forgot about that part you are right it should work fine if we are including the bottom of the page test.
also, the event is fired after the layout is updated.